File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.scala
/*
Solution
*/
implicit class StringExtensions(s: String) {
  def toAlternatingCase: String = {
    s.map {
      case c if c.isLower => c.toUpper
      case c if c.isUpper => c.toLower
      case c => c
    }
  }
}





/*
Sample Tests
*/
class AlternatingCaseSpec extends org.scalatest.flatspec.AnyFlatSpec {
  "toAlternatingCase" should "pass basic tests" in {
    val testCases = List(
      ("hello world", "HELLO WORLD"),
      ("HELLO WORLD", "hello world"),
      ("hello WORLD", "HELLO world"),
      ("HeLLo WoRLD", "hEllO wOrld"),
      ("12345", "12345"),
      ("1a2b3c4d5e", "1A2B3C4D5E"),
      ("Hello World".toAlternatingCase, "Hello World")
    )
    
    testCases.foreach {
      case (s, expected) => assertResult(expected, s"\nInput:\n  \"$s\"") (s.toAlternatingCase)
    }
  }
}

File: sol/solution/8 Kyu/56efc695740d30f963000557/Solution.go
/*
* Solution
*/
package kata
import "unicode"
func ToAlternatingCase(str string) string {
	runes := []rune(str)
	for i, r := range runes {
		if unicode.IsLower(r) {
			runes[i] = unicode.ToUpper(r)
		} else if unicode.IsUpper(r) {
			runes[i] = unicode.ToLower(r)
		}
	}
	return string(runes)
}





/*
* Sample Tests
*/
// Ginkgo BDD Testing Framework <http://onsi.github.io/ginkgo></http:>
// Gomega Matcher Library <http://onsi.github.io/gomega></http:>

package kata_test
import (
  . "github.com/onsi/ginkgo"
  . "github.com/onsi/gomega"
  . "codewarrior/kata"
)
var _ = Describe("Test Example", func() {
  It("basic tests", func() {
    Expect(ToAlternatingCase("hello world")).To(Equal("HELLO WORLD"))
    Expect(ToAlternatingCase("HELLO WORLD")).To(Equal("hello world"))
    Expect(ToAlternatingCase("hello WORLD")).To(Equal("HELLO world"))
    Expect(ToAlternatingCase("HeLLo WoRLD")).To(Equal("hEllO wOrld"))
    Expect(ToAlternatingCase("12345")).To(Equal("12345"))
    Expect(ToAlternatingCase("1a2b3c4d5e")).To(Equal("1A2B3C4D5E"))
    Expect(ToAlternatingCase("String.prototype.toAlternatingCase")).To(Equal("sTRING.PROTOTYPE.TOaLTERNATINGcASE"))
    Expect(ToAlternatingCase(ToAlternatingCase("Hello World"))).To(Equal("Hello World"))
    Expect(ToAlternatingCase("altERnaTIng cAsE")).To(Equal("ALTerNAtiNG CaSe"))
    Expect(ToAlternatingCase("ALTerNAtiNG CaSe")).To(Equal("altERnaTIng cAsE"))
    Expect(ToAlternatingCase("altERnaTIng cAsE <=> ALTerNAtiNG CaSe")).To(Equal("ALTerNAtiNG CaSe <=> altERnaTIng cAsE"))
    Expect(ToAlternatingCase("ALTerNAtiNG CaSe <=> altERnaTIng cAsE")).To(Equal("altERnaTIng cAsE <=> ALTerNAtiNG CaSe"))
  })
})